platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/FilterInjector.java

```diff
@@ -173,4 +187,51 @@ public void injectFilter(ServiceReference<ServletContext> serviceReference) {
         }
     }
 
+    public void addListener(ServiceReference<HttpSessionListener> sessionListener) {
+        if (sessionListener != null) {
+            Bundle refBundle = sessionListener.getBundle();
+            if (refBundle != null) {
+                BundleContext bundlectx = refBundle.getBundleContext();
+                if (bundlectx != null) {
+                    HttpSessionListener service = bundlectx.getService(sessionListener);
+                    if (service != null) {
+                        sessionListeners.add(service);
+                    }
+                }
+            }
+        }
+    }
+
+    public void removeListener(ServiceReference<HttpSessionListener> sessionListener) {
+        if (sessionListener != null) {
+            Bundle refBundle = sessionListener.getBundle();
+            if (refBundle != null) {
+                BundleContext bundlectx = refBundle.getBundleContext();
+                if (bundlectx != null) {
+                    HttpSessionListener service = bundlectx.getService(sessionListener);
+                    if (service != null) {
+                        sessionListeners.remove(service);
+                    }
+                }
+            }
+        }
+    }
+
+    private class WrapperListener implements HttpSessionListener {
```

Comment:
"Wrapper" is a pretty generic name. How about something more specific like event dispatcher or forwarder?